annotation/consume_trigger_test.go

```diff
@@ -0,0 +1,78 @@
+//  Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package annotation
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+)
+
+type ConsumingAnnotationTriggerTestSuite struct {
+	EqualsTestSuite
+}
```

Comment:
Maybe we can just have a `NewEqualsTestSuite` that takes the `interfaceName` and `initStructs` parameters, instead of having to inherit `EqualsTestSuite`?

Inheritance is second-class in Go, so I would try to avoid it when we can (plus having a constructor makes the code here simpler) :)

No strong preferences, of course.

(If we do change it here, we should also update the tests for produce trigger and key.

Reply:
Created issue #172 to perform this refactoring in a follow-up PR.